boefjes/plugins/kat_censys/normalize.py

```diff
@@ -86,3 +94,22 @@ def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterator[OOI
                     ooi=ip_port.reference, software=software_ooi.reference
                 )
 
+        if (
+            "http" in scan
+            and "response" in scan["http"]
+            and "headers" in scan["http"]["response"]
+        ):
+            headers = scan["http"]["response"]["headers"]
+            for header in headers:
```

Comment:
Nice one and clearly commented 👍 Just a few things that can be improved even more:
- You can use `str.startswith` to check the first character (e.g. `if not header.startswith('_'): ...`)
- Although the current statement (on line 105) evaluates to `True`, the `is (not)` operator is used to check wether two values point to the same object in memory, whilst `!=` compares the value or equality of two objects. Therefore the latter is preferable and the correct one to use in this situation
- I'm not sure yet what happens if the same OOI is yield twice, but it probably won't hurt. Do you know more about this @Lisser?
- Based on the code and comments we can infer that `headers` has the following type: `Dict[str, List[T]]`. So this part can be tidied up a little bit by iterating over a key-value pair rather looping over the keys and values separately, see pseudo code below

```python3
for header, values in headers.items():
    if header.startswith("_"):
        continue

    header = header.lower().replace("_", "-")

    for value in values:
        yield HTTPHeader(key=header, value=value, ...)
```

Reply:
Since the header will always be set (otherwise censys wouldn't return it), I think [0] will perform a bit better, but startsWith is more readable. I have changed it.
Currently, the certificates and headers do not appear in the GUI btw. I have no good way to link them to other objects. I also discussed this with @Lisser , but did not come to a decision yet. So that information is there on code level, but not yet in the GUI.